storage_service/locations/api/beta/remple/openapi.py

```diff
@@ -0,0 +1,1573 @@
+        fields = model_cls._meta.get_fields()
+        required_fields = []
+        for field in fields:
+            field_name = field.name
+            field_name, field_dict = {
+                ForeignKey: single_reference_field_dict,
+                OneToOneRel: single_reference_field_dict,
+                ManyToManyField: multi_reference_field_dict,
+                ManyToManyRel: multi_reference_field_dict,
+                ManyToOneRel: multi_reference_field_dict,
+            }.get(type(field), scalar_field_dict)(
+                field_name=field_name, field=field)
+            choices = get_choices(field)
+            if choices:
+                field_dict['enum'] = choices
+            if getattr(field, 'null', False) is True:
+                field_dict['nullable'] = True
+            default = get_default(field)
+            required, field_is_null = get_required(field, default)
+            default = get_default_for_type(field, default, field_is_null)
+            if default != NOT_PROVIDED:
+                field_dict['default'] = default
+            if required:
+                required_fields.append(field_name)
+            if not field_dict.get('type'):
+                logger.warning('{}.{} is of unknown type (class {})'.format(
+                    resource_name, field.name, type(field)))
+                continue
+            description = getattr(field, 'help_text', None)
+            if description:
+                if callable(description):
+                    description = description()
+                field_dict['description'] = str(description)
+            read_schema_properties[field_name] = field_dict
+        read_schema['properties'] = read_schema_properties
+        if required_fields:
+            read_schema['required'] = required_fields
+        return read_schema_name, read_schema
+
+    def get_api_version_slug(self):
+        try:
+            return self._get_api_version_slug(self.api_version)
+        except ValueError:
+            return self.api_version
+
+    @staticmethod
+    def _get_api_version_slug(version):
+        """Given a version number like 'X.Y.Z', return a slug representation of
+        it. E.g.,
+
+            >>> get_api_version_slug('3.0.0')
+            ... 'v3'
+            >>> get_api_version_slug('3.0.1')
+            ... 'v3_0_1'
+            >>> get_api_version_slug('3.0')
+            ... 'v3'
+            >>> get_api_version_slug('3.9')
+            ... 'v3_9'
+        """
+        parts = version.strip().split('.')
+        new_parts = []
+        for index, part in enumerate(parts):
+            part_int = int(part)
+            if part_int:
+                new_parts.append(part)
+            else:
+                parts_to_right = parts[index + 1:]
+                non_empty_ptr = [p for p in parts_to_right if int(p)]
+                if non_empty_ptr:
+                    new_parts.append(part)
+        return 'v{}'.format('_'.join(new_parts))
```

Comment:
The above is only needed if we want to support semver-compliant API versions, which may not be a good idea.